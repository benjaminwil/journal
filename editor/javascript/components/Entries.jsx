import React, {Component} from 'react';

import Entry from './Entry';

class Entries extends React.Component {
  render() {
    const entries = this.props.entries.map(entry => (
      <Entry key={entry.id} title={entry.title} content={entry.content} />
    ));

    return <div className="entries">{entries}</div>;
  }
}

export default Entries;
