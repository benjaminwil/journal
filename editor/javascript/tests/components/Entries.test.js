import React from 'react';
import renderer from 'react-test-renderer';
import {configure, shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

import Entries from '../../components/Entries.jsx';

function someEntries() {
  return [
    {
      id: '1',
      title: 'I walked for a while',
      datetime: {
        created: 'A million years ago',
        iso: {created: '2019-09-02T19:47:41.113Z'},
      },
      urls: {edit: '/entries/1/edit', view: '/entries/1'},
    },
    {
      id: '2',
      title: 'In my sleep',
      datetime: {
        created: 'A million years ago',
        iso: {created: '2019-09-02T19:47:41.113Z'},
      },
      urls: {edit: '/entries/2/edit', view: '/entries/2'},
    },
  ];
}

function entriesWrapper() {
  return mount(<Entries entries={someEntries()} />);
}

test('when no Entry is clicked, no EntryControls are displayed', () => {
  const wrapper = entriesWrapper();
  expect(wrapper.find('.controls')).not;
});

describe('an entry is selected and showing its EntryControls', () => {
  let entries;
  let firstEntry;
  let wrapper;

  beforeEach(() => {
    entries = someEntries();
    entries[0].active = true;
    wrapper = mount(<Entries entries={entries} />);
    firstEntry = wrapper.find('.entry').at(0);
  });

  test('the EntryControls are displayed', () => {
    expect(firstEntry.find('div.controls.enter-done'));
  });

  test('on click, EntryControls are hidden', () => {
    firstEntry.simulate('click');
    expect(firstEntry.find('div.controls.enter-done')).not;
  });

  test('when second entry is clicked, those EntryControls are shown instead', () => {
    const secondEntry = wrapper.find('.entry').at(1);
    secondEntry.simulate('click');
    expect(firstEntry.find('div.controls.enter-done')).not;
    expect(secondEntry.find('div.controls.enter-done'));
  });
});
