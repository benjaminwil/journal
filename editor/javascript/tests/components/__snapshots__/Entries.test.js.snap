// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Entries component displays many entries 1`] = `
<div
  className="entries"
>
  <article
    className="entry"
    onClick={[Function]}
  >
    <h1
      className="title"
    >
      I walked for a while
    </h1>
    <p />
    <div
      className="controls"
    >
      <a
        className="control edit"
        href="/entries/1/edit"
      >
        <span>
          Edit
        </span>
      </a>
      <a
        className="control view"
      >
        <span>
          View
        </span>
      </a>
      <button
        className="control delete"
        onClick={[Function]}
      >
        <span>
          Delete
        </span>
      </button>
    </div>
  </article>
  <article
    className="entry"
    onClick={[Function]}
  >
    <h1
      className="title"
    >
      In my sleep
    </h1>
    <p />
    <div
      className="controls"
    >
      <a
        className="control edit"
        href="/entries/2/edit"
      >
        <span>
          Edit
        </span>
      </a>
      <a
        className="control view"
      >
        <span>
          View
        </span>
      </a>
      <button
        className="control delete"
        onClick={[Function]}
      >
        <span>
          Delete
        </span>
      </button>
    </div>
  </article>
</div>
`;
